crates/oxc_transformer/src/regexp/mod.rs

```diff
@@ -0,0 +1,246 @@
+                            LookAroundAssertionKind::Lookbehind
+                                | LookAroundAssertionKind::NegativeLookbehind
+                        ) =>
+                {
+                    true
+                }
+                _ => false,
+            })
+        })
+    }
+
+    /// Check if the regular expression contains any unsupported syntax.
+    ///
+    /// Port from [esbuild](https://github.com/evanw/esbuild/blob/332727499e62315cff4ecaff9fa8b86336555e46/internal/js_parser/js_parser.go#L12667-L12800)
+    fn has_unsupported_regex_syntax_raw(&self, pattern: &str, flags: RegExpFlags) -> bool {
+        // Early return if no unsupported features-related plugins are enabled
+        if !(self.options.named_capture_groups
+            || self.options.unicode_property_escapes
+            || self.options.look_behind_assertions)
+        {
+            return false;
+        }
+
+        let is_unicode = flags.contains(RegExpFlags::U);
+        let mut paren_depth = 0;
+        let mut chars = pattern.chars().peekable();
+
+        while let Some(c) = chars.next() {
+            match c {
+                '[' => {
+                    while let Some(c) = chars.next() {
+                        if c == ']' {
+                            break;
+                        }
+                        if c == '\\' {
+                            chars.next();
+                        }
+                    }
+                }
+                '(' => {
+                    if matches!(chars.peek(), Some(&'?')) {
+                        chars.next();
+                        if matches!(chars.peek(), Some(&'<')) {
+                            chars.next();
+                            match chars.peek() {
+                                Some(&'!' | &'=') if self.options.look_behind_assertions => {
+                                    // (?<=) and (?<!)
+                                    return true;
+                                }
+                                _ => {
+                                    if self.options.named_capture_groups {
+                                        return chars.any(|c| c == '>');
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    paren_depth += 1;
+                }
+                ')' => {
+                    if paren_depth == 0 {
+                        return true;
+                    }
+                    paren_depth -= 1;
+                }
+                '\\' => {
+                    if self.options.unicode_property_escapes && is_unicode {
+                        if let Some(&next_char) = chars.peek() {
+                            // \p{ and \P{
+                            if (next_char == 'p' || next_char == 'P') && chars.nth(1) == Some('{') {
+                                return chars.any(|c| c == '}');
+                            }
```

Comment:
Same as above. Can this exit early when it shouldn't / can we remove the `chars.any(|c| c == '}')` check?

Reply:
No, I made a mistake here